File: case_itau_net_core/CaseItau.API/src/Domain/Entities/Fundo.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace CaseItau.API.src.Domain.Entities
{
    public class Fundo
    {
        [Key]
        [Required]
        [StringLength(20)]
        public string Codigo { get; set; }

        [Required]
        [StringLength(100)]
        public string Nome { get; set; }

        [Required]
        [StringLength(14)]
        public string Cnpj { get; set; }

        [Required]
        [ForeignKey("TipoFundo")]
        public int Codigo_Tipo { get; set; }

        [Column(TypeName = "NUMERIC")]
        public decimal? Patrimonio { get; set; }

        [JsonIgnore]
        public TipoFundo TipoFundo { get; set; }
    }
}

File: front_case_itau_angular/src/app/Fundo.ts
export class Fundo {
    codigo: string;
  
    nome: string;
  
    cnpj: string;
  
    codigo_Tipo: number;
  
    patrimonio?: number;
  
    constructor(codigo: string, nome: string, cnpj: string, codigo_Tipo: number, patrimonio?: number) {
      this.codigo = codigo;
      this.nome = nome;
      this.cnpj = cnpj;
      this.codigo_Tipo = codigo_Tipo;
      this.patrimonio = patrimonio;
    }
  }
  